eth/protocols/eth/handlers.go

```diff
@@ -274,6 +285,65 @@ func ServiceGetReceiptsQuery(chain *core.BlockChain, query GetReceiptsRequest) [
 	return receipts
 }
 
+// serviceGetReceiptsQuery69 assembles the response to a receipt query.
+// It does not send the bloom filters for the receipts
+func serviceGetReceiptsQuery69(chain *core.BlockChain, query GetReceiptsRequest) []rlp.RawValue {
+	// Gather state data until the fetch or network limits is reached
+	var (
+		bytes    int
+		receipts []rlp.RawValue
+	)
+	for lookups, hash := range query {
+		if bytes >= softResponseLimit || len(receipts) >= maxReceiptsServe ||
+			lookups >= 2*maxReceiptsServe {
+			break
+		}
+		// Retrieve the requested block's receipts
+		results := chain.GetRawReceiptsByHash(hash)
+		if results == nil {
+			if header := chain.GetHeaderByHash(hash); header == nil || header.ReceiptHash != types.EmptyRootHash {
+				continue
+			}
+		} else {
+			header := chain.GetHeaderByHash(hash)
+			if header.ReceiptHash != types.EmptyReceiptsHash {
+				body := chain.GetBody(hash)
```

Comment:
The other handles do not add body to cache (only the rlp-body. ). The GetBody uses `ReadBodyRLP` under the hood, perhaps you can/should use `chain.GetBodyRLP` instead? 

It will avoid putting the body object in cache. 
You can then either decode it into a body object, _or_ you can choose to iterate the rlp lists: the first one is the transactions. And you can decode one transaction at a time, or make some smart iterator which serves you the data you need. 

Serving this query is mighty expensive.

Reply:
GetBlodyRLP also caches it...
This query will be less expensive than before either way

Reply:
Rlp caches the rlp, not the objecr

This query will be a lot more expensive than before, as I see it. How do you reckon "less expensive"?